CMake build may fail when building with -j

When building with CMake and `-j4`, I get a failure in ROMIO:

```
configure: error: in `<charm location>/charm/netlrts-linux-x86_64/src/libs/ck-libs/ampi/romio-prefix/src/romio':
configure: error: C compiler cannot create executables
See `config.log' for more details
src/libs/ck-libs/ampi/CMakeFiles/romio.dir/build.make:108: recipe for target 'src/libs/ck-libs/ampi/romio-prefix/src/romio-stamp/romio-configure' failed
make[2]: *** [src/libs/ck-libs/ampi/romio-prefix/src/romio-stamp/romio-configure] Error 77
CMakeFiles/Makefile2:3071: recipe for target 'src/libs/ck-libs/ampi/CMakeFiles/romio.dir/all' failed
make[1]: *** [src/libs/ck-libs/ampi/CMakeFiles/romio.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

Looking at `config.log`indicates the error is caused by `g++: error: <charm location>/charm/netlrts-linux-x86_64/bin/../lib/conv-static.o: No such file or directory`. However, `conv-static.o` does exist and is in the correct location. This, coupled with the fact that the ROMIO output is interleaved with lines about building some files in `conv-core` makes me think this is due to a race from an unspecified dependency or something like that. (Note that I haven't been to reproduce this when building serially and if I go back and just run `make` in the build folder after encountering this failure, that works fine.)

--- a/src/libs/ck-libs/ampi/CMakeLists.txt
+++ b/src/libs/ck-libs/ampi/CMakeLists.txt
@@ -87,7 +87,7 @@ if(${CMK_AMPI_WITH_ROMIO})
 
     # Since we link a program with charmc as part of ROMIO's configure,
     # we need all libraries built at this point.
-    add_dependencies(romio moduleampi ampi-compat ck modulecompletion conv-partition conv-util hwloc conv-static)
+    add_dependencies(romio moduleampi ampi-compat ck modulecompletion conv-partition conv-util hwloc conv-static conv-core)
     if(${CMK_CAN_LINK_FORTRAN})
         add_dependencies(romio moduleampif)
     endif()

--- a/src/libs/ck-libs/ampi/CMakeLists.txt
+++ b/src/libs/ck-libs/ampi/CMakeLists.txt
@@ -87,7 +87,7 @@ if(${CMK_AMPI_WITH_ROMIO})
 
     # Since we link a program with charmc as part of ROMIO's configure,
     # we need all libraries built at this point.
-    add_dependencies(romio moduleampi ampi-compat ck modulecompletion conv-partition conv-util hwloc)
+    add_dependencies(romio moduleampi ampi-compat ck modulecompletion conv-partition conv-util hwloc conv-static)
     if(${CMK_CAN_LINK_FORTRAN})
         add_dependencies(romio moduleampif)
     endif()

